GeometryCore/src/Polygon.cpp

```diff
@@ -58,13 +70,55 @@ namespace geom_utils
     // Calculate and return convex hull for the polygon
     Polygon Polygon::convexHull() const 
     { 
-        return {}; 
+        //if (points.size() < 3) return; //can't bild a polygon from 2 points or less //idk how to do it
+        Polygon convexHull;
+        int pos_of_the_most_right_point = 0;
+
+        for (int i = 1; i < points.size(); i++) if (points[i].x < points[pos_of_the_most_right_point].x) pos_of_the_most_right_point = i; //idk how to do it with stl
```

Comment:
It is ok.
But you can also use `std::min_element()` which will explicitly state our intention of searching for some minimum element in a range. As a predicate for `std::min_element()` you can use your comparison of `x` component.

Reply:
oh, it looks great. i will implement it